### PR TITLE
[TDL-23979] Fix the existing schema and discovery

### PR DESCRIPTION
**Pull request by shantanu73**
# Description of change
1. Fixed Discovery by adding "valid-replication-keys" parameter in catalog
2. Schema changes as per API documentation

# Manual QA steps
 - Ran discovery and validated schema in catalog.json
 
# Risks
 - Incorrect schema may cause integration tests to fail.
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_outreach/schemas/accounts.json [hunk 1]
```diff
@@ -7,20 +7,18 @@
...
       ]
     },
-    "companyType": {
+    "buyerIntentScore": {
       "type": [
-        "null",
-        "string"
+        "number"
       ]
     },
```
Comment (RushiT0122) [inline tap_outreach/schemas/accounts.json:14]:
```suggestion
    "buyerIntentScore": {
      "type": [
        "null",
        "number"
      ]
    },
```

File: tap_outreach/discover.py [hunk 2]
```diff
@@ -28,9 +28,11 @@ def get_schemas():
             singer.resolve_schema_references(schema, refs)
 
         replication = STREAM_CONFIGS[stream_name]['replication']
+        replication_keys = STREAM_CONFIGS[stream_name].get('filter_field', None)
```
Comment (RushiT0122) [inline tap_outreach/discover.py:31]: Is `replication_keys` a list or a single value? If single value can we change it to `replication_key`?

Comment (shantanu73) [reply tap_outreach/discover.py]: Fixed in latest commit.

File: tap_outreach/schemas/accounts.json (hunk 1 above)
Comment (shantanu73) [reply tap_outreach/schemas/accounts.json]: Fixed in latest commit.

**Review by RushiT0122 — APPROVED** [review]

**Pull request merged**